compiler/resolution/lifetime.cpp

```diff
@@ -953,109 +1312,136 @@ bool EmitLifetimeErrorsVisitor::enterCallExpr(CallExpr* call) {
   if (call->id == debugLifetimesForId)
     gdbShouldBreakHere();
 
+  // Ignore calls that de-temping allows us to ignore
+  if (lifetimes->callsToIgnore.count(call))
+    return false;
+
   if (isAnalyzedMoveOrAssignment(call)) {
```

Comment:
This function is pretty long. How can we organize the code? Ex.

    checkXXX(call);
    checkYYY(call);
    checkZZZ(call);

?